training/src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,137 @@
+    public Iterable findAll() {
+        return userRepository.findAll();
+    }
+
+    @GetMapping("/users/{id}")
+    public Users findOne(@PathVariable Long id) {
+        return userRepository.findById(id)
+            .orElseThrow(UserNotFoundException::new);
+    }
+
+    private void generateException(String exception) {
+
+        switch (exception) {
+            case "userNotFoundException":
+                try {
+                    throw new UserNotFoundException();
+                } catch (UserNotFoundException e) {
+                    e.printStackTrace();
+                }
+                break;
+            case "bookIdMismatchException": {
+                try {
+                    throw new BookIdMismatchException();
+                } catch (BookIdMismatchException e) {
+                    e.printStackTrace();
+                }
+            }
+            break;
+            case "bookAlreadyOwnedException": {
+                try {
+                    throw new BookAlreadyOwnedException();
+                } catch (BookAlreadyOwnedException e) {
+                    e.printStackTrace();
+                }
+            }
+            break;
+
+            case "bookNotFoundException": {
+                try {
+                    throw new BookNotFoundException();
+                } catch (BookNotFoundException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+
+    @DeleteMapping("/users/{id}")
+    public void delete(@PathVariable Long id) {
+        Optional<Users> userToDelete = userRepository.findById(id);
+        if (userToDelete.isPresent()) {
+            userToDelete.get().getBooks().removeAll(userToDelete.get().getBooks());
+            userRepository.delete(userToDelete.get());
+        } else {
+            this.generateException("userNotFoundException");
+        }
+    }
+
+
+    @PostMapping("/createUser")
+    @ResponseStatus(HttpStatus.CREATED)
+    public Users create(@RequestBody Users users) {
+        Users newUsers = new Users(users.getUsername(), users.getName(), users.getBirthdate());
+        return userRepository.save(newUsers);
+    }
+
+    @PutMapping("/users/{id}")
+    public Users updateUser(@RequestBody Users users, @PathVariable Long id) {
+        if (users.getId() != id) {
+            this.generateException("userNotFoundException");
+        }
+        userRepository.findById(id)
+            .orElseThrow(UserNotFoundException::new);
+        return userRepository.save(users);
+    }
+
+    @PutMapping("/users/{id_user}/book/{id_book}")
+    public Users updateBookOfUser(@PathVariable String id_user, @PathVariable String id_book) {
+        Optional<Users> user = userRepository.findById(Long.parseLong(id_user));
+        if (user.isPresent()) {
+            Optional<Book> bookToAssing = bookRepository.findById(Long.parseLong(id_book));
+            if (bookToAssing.isPresent()) {
+                if (user.get().getBooks().stream()
```

Comment:
It would be easier if the `user.addBook` method was the responsible of checking if the user already owns the book.  It's the responsibility of the user to know which books he contains, not from the controller.
Besides, as a rule, the controller must contain the less "business logic" as possible.